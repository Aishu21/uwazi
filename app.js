require("node-jsx").install();
var express = require('express');
var app = express();

var react = require('react');
var router = require('react-router');

var renderToString = require('react-dom/server').renderToString;
var routes = require('./js/routes.jsx');
var match = require('react-router').match;
var RoutingContext = require('react-router').RoutingContext;
//var reactApp = React.createFactory(require('./js/react_welcome.js'));
//var layout = require('./js/layout.jsx');

app.get('/', function (req, res) {

  match({ routes: routes, location: req.url }, function(error, redirectLocation, renderProps) {
    res.status(200).send(renderToString(new RoutingContext(renderProps)))
  });

  //res.render(__dirname+'/views/index.ejs', {
    //content: content
  //});

  //router.run(routes, '/', function (Handler) {
    //var content = react.renderToString(react.createElement(Handler));
    //res.render(__dirname+'/views/index.ejs', {
      //content: content
    //});
  //});
})

//app.get('/', function (req, res) {
  //var reactHtml = React.renderToString(ReactApp({}));
  //res.render(__dirname+'/views/welcome.ejs', {reactOutput: reactHtml});
//});

var server = app.listen(3000, function () {
  var host = server.address().address;
  var port = server.address().port;

  console.log('Example app listening at http://%s:%s', host, port);
});
