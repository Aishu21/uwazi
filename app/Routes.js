import React from 'react'
import { Route, IndexRoute } from 'react-router'
import App from './components/App'
import Home from './components/Home'
import Users from './components/Users'
import Login from './components/Login'
import NoMatch from './components/NoMatch'

export default (
	<Route path='/' component={App}>
		<IndexRoute component={Home} />
		<Route path='users' component={Users} />
		<Route path='login' component={Login} />
		<Route path="*" component={NoMatch} />
	</Route>
);
