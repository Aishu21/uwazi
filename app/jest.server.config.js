module.exports = {
  name: 'server',
  displayName: 'Server',
  testMatch: ['**/api/**/specs/*spec.(j|t)s?(x)', '**/shared/**/specs/*spec.(j|t)s?(x)'],
  testEnvironment: 'node',
  setupFilesAfterEnv: ['<rootDir>/setUpJestServer.js'],
  collectCoverageFrom: ['app/**/*.{js,ts}'],
  coverageDirectory: 'coverage',
  coverageReporters: ['none'],
};
