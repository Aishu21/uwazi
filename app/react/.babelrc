{
  "presets": ["es2015", "react"],
  "plugins": [
    ["babel-plugin-module-alias", [
      { "src": "./app/react", "expose": "~" },
      { "src": "./app/shared", "expose": "shared" },
    ]],
    "transform-class-properties",
    "add-module-exports"
  ]
}
