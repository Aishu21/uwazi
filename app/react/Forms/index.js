import Select, {SelectField} from './components/Select';
import DatePicker, {DatePickerField} from './components/DatePicker';
import DateRange from './components/DateRange';
import FormField from './components/FormField';
import FormGroup from './components/FormGroup';
import MultiSelect from './components/MultiSelect';

export {
  Select,
  SelectField,
  FormField,
  FormGroup,
  DatePicker,
  DatePickerField,
  DateRange,
  MultiSelect
};
