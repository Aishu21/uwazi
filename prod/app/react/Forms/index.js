"use strict";Object.defineProperty(exports, "__esModule", { value: true });Object.defineProperty(exports, "Select", { enumerable: true, get: function () {return _Select.default;} });Object.defineProperty(exports, "DatePicker", { enumerable: true, get: function () {return _DatePicker.default;} });Object.defineProperty(exports, "Captcha", { enumerable: true, get: function () {return _Captcha.default;} });Object.defineProperty(exports, "DateRange", { enumerable: true, get: function () {return _DateRange.default;} });Object.defineProperty(exports, "FormGroup", { enumerable: true, get: function () {return _FormGroup.default;} });Object.defineProperty(exports, "MultiSelect", { enumerable: true, get: function () {return _MultiSelect.default;} });Object.defineProperty(exports, "MarkDown", { enumerable: true, get: function () {return _MarkDown.default;} });Object.defineProperty(exports, "Nested", { enumerable: true, get: function () {return _Nested.default;} });Object.defineProperty(exports, "NestedMultiselect", { enumerable: true, get: function () {return _NestedMultiselect.default;} });Object.defineProperty(exports, "MultiDate", { enumerable: true, get: function () {return _MultiDate.default;} });Object.defineProperty(exports, "MultiDateRange", { enumerable: true, get: function () {return _MultiDateRange.default;} });Object.defineProperty(exports, "DropdownList", { enumerable: true, get: function () {return _DropdownList.default;} });Object.defineProperty(exports, "IconSelector", { enumerable: true, get: function () {return _IconSelector.default;} });Object.defineProperty(exports, "Numeric", { enumerable: true, get: function () {return _Numeric.default;} });Object.defineProperty(exports, "NumericRange", { enumerable: true, get: function () {return _NumericRange.default;} });Object.defineProperty(exports, "NumericRangeSlide", { enumerable: true, get: function () {return _NumericRangeSlide.default;} });Object.defineProperty(exports, "RadioButtons", { enumerable: true, get: function () {return _RadioButtons.default;} });Object.defineProperty(exports, "Switcher", { enumerable: true, get: function () {return _Switcher.default;} });Object.defineProperty(exports, "Geolocation", { enumerable: true, get: function () {return _Geolocation.default;} });Object.defineProperty(exports, "LinkField", { enumerable: true, get: function () {return _LinkField.default;} });Object.defineProperty(exports, "FormValue", { enumerable: true, get: function () {return _FormValue.default;} });var _Select = _interopRequireDefault(require("./components/Select"));
var _DatePicker = _interopRequireDefault(require("./components/DatePicker"));
var _Captcha = _interopRequireDefault(require("./components/Captcha"));
var _DateRange = _interopRequireDefault(require("./components/DateRange"));
var _FormGroup = _interopRequireDefault(require("./components/FormGroup"));
var _MultiSelect = _interopRequireDefault(require("./components/MultiSelect"));
var _MarkDown = _interopRequireDefault(require("./components/MarkDown"));
var _Nested = _interopRequireDefault(require("./components/Nested"));
var _NestedMultiselect = _interopRequireDefault(require("./components/NestedMultiselect"));
var _MultiDate = _interopRequireDefault(require("./components/MultiDate"));
var _MultiDateRange = _interopRequireDefault(require("./components/MultiDateRange"));
var _DropdownList = _interopRequireDefault(require("./components/DropdownList"));
var _IconSelector = _interopRequireDefault(require("./components/IconSelector"));
var _Numeric = _interopRequireDefault(require("./components/Numeric"));
var _NumericRange = _interopRequireDefault(require("./components/NumericRange"));
var _NumericRangeSlide = _interopRequireDefault(require("./components/NumericRangeSlide"));
var _RadioButtons = _interopRequireDefault(require("./components/RadioButtons"));
var _Switcher = _interopRequireDefault(require("./components/Switcher"));
var _Geolocation = _interopRequireDefault(require("./components/Geolocation"));
var _LinkField = _interopRequireDefault(require("./components/LinkField"));
var _FormValue = _interopRequireDefault(require("./components/FormValue"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}