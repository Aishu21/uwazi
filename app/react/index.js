import React from 'react';
import ReactDOM from 'react-dom';
import {browserHistory} from 'react-router';
import {Router} from 'react-router';
import Routes from './Routes';
import {Provider} from 'react-redux';
import CustomProvider from './App/Provider';
import store from './store';
import {AppContainer} from 'react-hot-loader';

const App = () => {
  return (
    <Provider store={store()}>
      <CustomProvider>
        <Router history={browserHistory}>{Routes}</Router>
      </CustomProvider>
    </Provider>
  );
};

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <App/>
    </AppContainer>,
    document.getElementById('root')
  );
};

render();
if (module.hot) {
  module.hot.accept();
}

import './sockets';

