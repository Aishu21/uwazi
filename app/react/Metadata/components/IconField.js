import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { IconSelector } from 'app/ReactReduxForms';
import { actions } from 'app/Metadata';
import { FormValue } from 'app/Forms';
import ToggleDisplay from 'app/Layout/ToggleDisplay';
import { bindActionCreators } from 'redux';
import { Translate } from 'app/I18N';
import { Icon } from 'UI';

export const IconField = ({ model, removeIcon }) => (
  <FormValue model={`${model}.icon`}>{
    (icon = {}) => (
      <ToggleDisplay
        showLabel={<span><Translate>show icon</Translate><Icon icon="eye" /></span>}
        hideLabel={<span>{<Translate>hide icon</Translate>}<Icon icon="eye-slash" /></span>}
        onHide={() => removeIcon(`${model}.icon`)}
        open={!!icon._id}
      >
        <ul className="search__filter">
          <li><label><Translate>Icon</Translate> / <Translate>Flag</Translate></label></li>
          <li className="wide">
            <IconSelector model=".icon"/>
          </li>
        </ul>
      </ToggleDisplay>
    )}
  </FormValue>
);

IconField.propTypes = {
  model: PropTypes.string.isRequired,
  removeIcon: PropTypes.func.isRequired,
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ removeIcon: actions.removeIcon }, dispatch);
}

export default connect(null, mapDispatchToProps)(IconField);
