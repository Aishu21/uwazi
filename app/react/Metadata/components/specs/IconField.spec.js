import React from 'react';

import { FormValue } from 'app/Forms';
import { shallow } from 'enzyme';
import ToggleDisplay from 'app/Layout/ToggleDisplay';

import { IconField } from '../IconField';

describe('IconField', () => {
  let props;
  beforeEach(() => {
    props = {
      removeIcon: jasmine.createSpy('removeIcon')
    };
  });
  it('should render IconSelector with toggleDisplay', () => {
    const component = shallow(<IconField {...props} model="model.value"/>).find(FormValue).prop('children')();
    expect(component).toMatchSnapshot();
  });

  it('should be open when has value', () => {
    const component = shallow(<IconField {...props} model="model.value"/>).find(FormValue).prop('children')({ _id: 'id' });
    expect(component).toMatchSnapshot();
  });

  it('should call removeIcon on hide', () => {
    const component = shallow(<span>{shallow(<IconField {...props} model="model.value"/>).find(FormValue).prop('children')('value')}</span>);
    component.find(ToggleDisplay).props().onHide();
    expect(props.removeIcon).toHaveBeenCalledWith('model.value.icon');
  });
});
