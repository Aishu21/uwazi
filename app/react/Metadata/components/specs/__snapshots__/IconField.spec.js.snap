// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconField should be open when has value 1`] = `
<ToggleDisplay
  hideLabel={
    <span>
      <Connect(Translate)>
        hide icon
      </Connect(Translate)>
      <Unknown
        icon="eye-slash"
      />
    </span>
  }
  onHide={[Function]}
  open={true}
  showLabel={
    <span>
      <Connect(Translate)>
        show icon
      </Connect(Translate)>
      <Unknown
        icon="eye"
      />
    </span>
  }
>
  <ul
    className="search__filter"
  >
    <li>
      <label>
        <Connect(Translate)>
          Icon
        </Connect(Translate)>
         / 
        <Connect(Translate)>
          Flag
        </Connect(Translate)>
      </label>
    </li>
    <li
      className="wide"
    >
      <IconSelector
        model=".icon"
      />
    </li>
  </ul>
</ToggleDisplay>
`;

exports[`IconField should render IconSelector with toggleDisplay 1`] = `
<ToggleDisplay
  hideLabel={
    <span>
      <Connect(Translate)>
        hide icon
      </Connect(Translate)>
      <Unknown
        icon="eye-slash"
      />
    </span>
  }
  onHide={[Function]}
  open={false}
  showLabel={
    <span>
      <Connect(Translate)>
        show icon
      </Connect(Translate)>
      <Unknown
        icon="eye"
      />
    </span>
  }
>
  <ul
    className="search__filter"
  >
    <li>
      <label>
        <Connect(Translate)>
          Icon
        </Connect(Translate)>
         / 
        <Connect(Translate)>
          Flag
        </Connect(Translate)>
      </label>
    </li>
    <li
      className="wide"
    >
      <IconSelector
        model=".icon"
      />
    </li>
  </ul>
</ToggleDisplay>
`;
